### PR TITLE
[TASK] Add section on deep linking

### LINKED ISSUE #51: Add "deep linking" article for TYPO3 v11
Since TYPO3 v11 LTS deep linking is available for editors - see section "Deep Links in the TYPO3 Backend" at https://typo3.org/article/typo3-v11-lts-warp-speed. Write an article about it by e.g. 

- adding a screenshot with arrow to the new link button
- re-use the explanation of the above mentioned article
- explain the difference between copying the deep link URL from the browser address bar or the new TYPO3 backend button:
  Button is providing shortest, cleanest form of deep linking URL reduced by any user tracks like the URL query param `&returnUrl`.

### PR DESCRIPTION
**Pull request by tomwarwick**
Resolves #51

This commit adds a new section on deep linking
to the editors guide. The intent of this section is
to provide editors with an overview of what deep
linking is and how it can be used to better collaborate
with other users.

[+} Update global menu
[+] Add new deep linking chapter
[+] Add manual images

### CODE REVIEW

File: Documentation/DeepLinking/Index.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+on the `About Us` page:
+
+.. figure:: ../Images/ManualScreenshots/DeepLinking/DeepLinkingContentElementLink.png
+  :alt: Creating a share link to a content element
+  :class: with-border
+
+Linking to backend modules
+--------------------------
+
+It is possible to create links to places within Modules. This example would crea
```
Comment (sypets) [inline Documentation/DeepLinking/Index.rst:53]: Something is missing here?

File: Documentation/DeepLinking/Index.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  :alt: Creating a link to the backend Sites module
+  :class: with-border
+
+.. _deeplinking-bestpractices:
+
+URLs and deep links
+===================
+
+When creating links to anything in the backend, it is recommended that the
+`Share` button is used to generate the link. Copying the URL from the browser
```
Comment (sypets) [inline Documentation/DeepLinking/Index.rst:65]: Above, _Share_ is also used but without the backticks. I am not sure what would be preferred here. If we refer to something where the text is also used in the UI (which is not the case here) we can use interpreted text roles guilabel, but that is not the case here. 

```
:guilabel:`Share`
```

see also https://docs.typo3.org/m/typo3/docs-how-to-document/master/en-us/WritingReST/InlineCode.html

I am unsure of the convention here, but I would change it so it is the same in all cases.

(I just saw so much (perceived) overuse of inline markup that I now tend to not use it ... but that is a question of general conventions).

**Review by sypets — CHANGES_REQUESTED** [review]

File: Documentation/DeepLinking/Index.rst (hunk 1 above)
Comment (tomwarwick) [reply Documentation/DeepLinking/Index.rst]: Resolved - thank you.

File: Documentation/DeepLinking/Index.rst (hunk 2 above)
Comment (tomwarwick) [reply Documentation/DeepLinking/Index.rst]: Resolved.

**Review by linawolf — APPROVED** [review]

**Review by susannemoog — APPROVED** [review]

**Pull request merged**